src/core/Form/Select/SingleSelect/SingleSelect.tsx

```diff
@@ -239,11 +244,13 @@ class BaseSingleSelect<T> extends Component<
                 item.uniqueItemId === prevState.selectedItem?.uniqueItemId,
             );
       const resolvedInputValue = selectedItemChanged
-        ? selectedItem?.labelText || ''
+        ? selectedItem !== null
+          ? selectedItem?.labelText || prevState.filterInputValue
+          : ''
```

Comment:
Just a note: This starts to be a little difficult to read with the nested if-logic

Reply:
Might be a good idea to refactor into a slightly more readable form.

One potential solution (still not super clear thought):

```js
      let resolvedInputValue = prevState.filterInputValue;
      if (selectedItemChanged) {
        resolvedInputValue = selectedItem
          ? selectedItem.labelText || prevState.filterInputValue
          : '';
      } else {
        const matchingItem = propItems.find(
          (item) => item.uniqueItemId === prevState.selectedItem?.uniqueItemId,
        );
        if (matchingItem) {
          resolvedInputValue =
            matchingItem.labelText || prevState.filterInputValue;
        }
      }
```